src/runtime/CMakeLists.txt

```diff
@@ -144,7 +143,8 @@ add_library(Halide_initmod OBJECT)
 
 set(CXX_WARNING_FLAGS -Wall -Werror -Wno-unused-function -Wcast-qual)
 set(RUNTIME_CXX_FLAGS -O3
-    -fno-vectorize -ffreestanding -fno-blocks -fno-exceptions -fno-unwind-tables -fpic
+    -fno-vectorize -ffreestanding -fno-blocks -fno-exceptions -fno-unwind-tables
+    #-fpic
```

Comment:
Not entirely sure how to fix this in an elegant way... `-fpic` can't be used with `x64-windows`...

Reply:
Alas, I don't know enough about the issues to have a useful comment.

Reply:
@alexreinking probably knows the right thing to do

Reply:
_Why_ can't `-fpic` be used with `x64-windows`? Regardless of how you implement it, that merits a comment. As far as the CMake hacking goes, I would do something like this:

```
foreach (j IN ITEMS 32 64)
  set(fpic -fpic)
  if (${j} EQUAL 32)
  ...
  else ()
    if (${i} MATCHES "windows_.*")
      set(fpic "") # comment explaining why this is
  ...
  foreach (SUFFIX IN ITEMS "" "_debug")
    ...
    add_custom_command(OUTPUT "${LL}"
                       COMMAND clang ${RUNTIME_CXX_FLAGS} ${fpic}
```

Reply:
Yeah, it's bizarre that `-fpic` can't be used, but clang wasn't happy:
```
clang : error : unsupported option '-fpic' for target 'x86_64-pc-windows-msvc'
```

Reply:
Ah, after a little research, it seems that PIC doesn't apply to Windows at all because DLL function addresses are fixed, so there's no point in including it. Then we should set that `fpic` variable to empty when the target is windows and to `-fpic` otherwise.

Reply:
Yeah, I think clang actually issues a warning saying that -fpic is not needed and will be ignored, but since we treat warnings as errors, this ends up interrupting the build as an error.